File: CodeChef/Self Learning/Basic Programming Concepts/Two_Dishes.py
# CodeChef - TWODISH - Two Dishes
# https://www.codechef.com/LP1TO201/problems/TWODISH

import sys
input = sys.stdin.readline

T = int(input())
for _ in range(T):
    N, A, B, C = map(int, input().split())
    ans = "YES" if A+C >= N and B >= N else "NO"
    print(ans)

File: CodeChef/Self Learning/Basic Programming Concepts/Two_Dishes.cpp
// CodeChef - TWODISH - Two Dishes
// https://www.codechef.com/LP1TO201/problems/TWODISH

#include "bits/stdc++.h"
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int T;
    cin >> T;
    while(T--){
        int N, A, B, C;
        cin >> N >> A >> B >> C;

        string ans = A+C >= N && B >= N ? "YES" : "NO";
        cout << ans << "\n";
    }

}
